### PR TITLE
added cluster name specificity to template names resolves #595

### LINKED ISSUE #595: Template conflict with multiple instances of teraslice on one ES cluster
Teraslice uses the `analytics_template` and `state_template` templates to enforce mappings onto it's backend support indices.  If you have multiple versions of teraslice that use the same backend elasticsearch and these require different templates then there will be a conflict.

The template names and their corresponding matching strings should probably include the teraslice instance name to disambiguate.

### PR DESCRIPTION
**Pull request by jsnoble**

### CODE REVIEW

File: lib/cluster/storage/backends/mappings/analytics.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-    "template": "*__analytics*",
+    "template": "__analytics*",
```
Comment (godber) [inline lib/cluster/storage/backends/mappings/analytics.json]: These are the strings used to match the indices that the mappings get applied to right?  Don't these have to only apply to the indices specific to the cluster?  That is shouldn't these have the form like `${clusterName}__analytics*`?  Same goes for `logs` and `state`.

Or is this somehow being handled elsewhere?

Comment (jsnoble) [reply lib/cluster/storage/backends/mappings/analytics.json]: yes in the elasticsearch_store  since all the stores use it but one

Comment (godber) [reply lib/cluster/storage/backends/mappings/analytics.json]: Oh, OK, I see it now.

**Comment by godber** [comment]
Integration tests pass

screenshot_2017-11-16_07-02-50

**Comment by godber** [comment]
The templates have the cluster name in them now:

```
curl -Ss localhost:49200/_template | jq . | grep template
  "teracluster_state_template": {
    "template": "teracluster__state*",
  "teracluster_analytics_template": {
    "template": "teracluster__analytics*",
```

**Pull request merged**